File: src/main/java/pl/lodz/p/ias/io/powiadomienia/announcement/Announcement.java
package pl.lodz.p.ias.io.powiadomienia.announcement;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.validator.constraints.Length;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Announcement {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    @Length(min = 1, max = 1024)
    private String message;
    @Length(min = 1, max = 64)
    private String title;
    private AnnouncementType type;
}

File: front/src/powiadomienia/types/announcement.ts

export enum AnnouncementType {
    INFORMATION = "INFORMATION",
    WARNING = "WARNING",
  }


export interface createAnnouncement{
  message:string;
  title:string;
  type:AnnouncementType
}

export default interface Announcement{
    message:string;
    type:AnnouncementType,
    title:string,
    id:number
}
